asapdiscovery-alchemy/asapdiscovery/alchemy/predict.py

```diff
@@ -0,0 +1,817 @@
+        if "DDG (kcal/mol)" in column:
+            df = df.rename(
+                columns={
+                    column: f"{column.replace('DDG (kcal/mol)', 'relative pIC50')}"
+                }
+            )
+        elif "DG (kcal/mol)" in column:
+            df = df.rename(
+                columns={column: f"{column.replace('DG (kcal/mol)', 'pIC50')}"}
+            )
+        elif "uncertainty (kcal/mol)" in column:
+            df = df.rename(
+                columns={
+                    column: f"{column.replace('uncertainty (kcal/mol)', 'pIC50 uncertainty')}"
+                }
+            )
+        elif "prediction error (kcal/mol)" in column:
+            df = df.rename(
+                columns={
+                    column: f"{column.replace('prediction error (kcal/mol)', 'prediction error (pIC50)')}"
+                }
+            )
+
+    return df
+
+
+def add_smiles_to_df(dataframe: pd.DataFrame, ligands: list) -> pd.DataFrame:
+    """
+    Given a wrangled DF containing either `label` (absolute) or `labelA` and `labelB` (relative),
+    add SMILES for each row.
+
+    Args:
+        dataframe: The pandas dataframe we should add the smiles to.
+        ligands: The list of ligands from which we should extract the smiles based on matching by name.
+    Returns:
+        The dataframe with added smiles column(s)
+    """
+    # get the experimental data for SMILES per compound name.
+    expt_smiles = {ligand.name: ligand.smiles for ligand in ligands}
+
+    # get the SMILES. Relative dataframe is a bit more involved than absolute.
+    if "labelA" in dataframe.columns:
+        smiles_a = []
+        smiles_b = []
+        for labelA, labelB in dataframe[["labelA", "labelB"]].values:
+            smiles_a.append(expt_smiles[labelA])
+            smiles_b.append(expt_smiles[labelB])
+        dataframe["SMILES_A"] = smiles_a
+        dataframe["SMILES_B"] = smiles_b
+
+    elif "label" in dataframe.columns:
+        smiles = [expt_smiles[label] for label in dataframe["label"].values]
+        dataframe["SMILES"] = smiles
+
+    return dataframe
+
+
+def extract_experimental_data(
+    reference_csv: str, assay_units: Literal["pIC50", "IC50"]
+) -> dict[str, tuple[unit.Quantity, unit.Quantity]]:
+    """
+    Extract the experimental data from the given csv file, this assumes the csv has been downloaded from cdd.
+    Where the molecule identifier is under column 'Molecule Name' and the experimental data is pIC50 / IC50
+    TODO make more general
+
+    Args:
+        reference_csv: The name of the csv file with the experimental data
+        assay_units: The assay units of 'pIC50' or 'IC50' that the experimental data is given in.
+
+    Returns:
+        A dictionary of molecule names and tuples of
+        experimental data and its associated uncertainty converted to Gibbs free energy in kcal/mol.
+    """
+    experimental_data = {}
+    assay_tag = assay_units + "_Mean"
```

Comment:
this won't be enough to catch column names here, unfortunately. "*_Mean" only occurs when experiments were run in (at least) duplicate, which is only sometimes the case. We'll need to add a CLI flag that allows the user to feed in the string instead, because the experiment string can vary per target/assay.

Reply:
so my suggestion is to keep line 258, but if it isn't found: suggest that the user supplies the correct string.

Reply:
resolved in https://github.com/choderalab/asapdiscovery/pull/711/commits/b83a23d95a0e64fdefbaf89bbe9906fd737832a3